ssz/src/main/java/tech/pegasys/artemis/util/backing/tree/TreeNode.java

```diff
@@ -153,6 +153,15 @@ default TreeNode updated(long target, Function<TreeNode, TreeNode> nodeUpdater)
    */
   TreeNode updated(long generalizedIndex, Function<TreeNode, TreeNode> nodeUpdater);
 
+  /** Updates the tree in a batch */
+  default TreeNode updated(TreeUpdates newNodes) {
```

Comment:
I would leave this as a basic method implementation for new potential interface implementations. Also it might serve some documentation purpose - i.e. illustrate what the method actually does without any optimizations